libsrc/xml.cpp

```diff
@@ -790,430 +807,443 @@ void append_optional_three_dimensional_float(pugi::xml_node& n, const char* chil
-  }
-  bool operator!=(const GradientDirection &lhs, const GradientDirection &rhs) {
-      return !(rhs == lhs);
-  }
-  }
+}
+
+bool operator==(const IsmrmrdHeader &lhs, const IsmrmrdHeader &rhs) {
+    return lhs.version == rhs.version &&
+           lhs.subjectInformation == rhs.subjectInformation &&
+           lhs.studyInformation == rhs.studyInformation &&
+           lhs.measurementInformation == rhs.measurementInformation &&
+           lhs.acquisitionSystemInformation == rhs.acquisitionSystemInformation &&
+           lhs.experimentalConditions == rhs.experimentalConditions &&
+           lhs.encoding == rhs.encoding &&
+           lhs.sequenceParameters == rhs.sequenceParameters &&
+           lhs.userParameters == rhs.userParameters &&
+           lhs.waveformInformation == rhs.waveformInformation;
+}
+bool operator!=(const IsmrmrdHeader &lhs, const IsmrmrdHeader &rhs) {
+    return !(rhs == lhs);
+}
+bool operator==(const SubjectInformation &lhs, const SubjectInformation &rhs) {
+    return lhs.patientName == rhs.patientName &&
+           lhs.patientWeight_kg == rhs.patientWeight_kg &&
+           lhs.patientID == rhs.patientID &&
+           lhs.patientBirthdate == rhs.patientBirthdate &&
+           lhs.patientGender == rhs.patientGender;
+}
+bool operator!=(const SubjectInformation &lhs, const SubjectInformation &rhs) {
+    return !(rhs == lhs);
+}
+bool operator==(const StudyInformation &lhs, const StudyInformation &rhs) {
+    return lhs.studyDate == rhs.studyDate &&
+           lhs.studyTime == rhs.studyTime &&
+           lhs.studyID == rhs.studyID &&
+           lhs.accessionNumber == rhs.accessionNumber &&
+           lhs.referringPhysicianName == rhs.referringPhysicianName &&
+           lhs.studyDescription == rhs.studyDescription &&
+           lhs.studyInstanceUID == rhs.studyInstanceUID &&
+           lhs.bodyPartExamined == rhs.bodyPartExamined;
+}
+bool operator!=(const StudyInformation &lhs, const StudyInformation &rhs) {
+    return !(rhs == lhs);
+}
+bool operator<(const MeasurementDependency &lhs, const MeasurementDependency &rhs) {
+    return lhs.dependencyType < rhs.dependencyType || (lhs.dependencyType == rhs.dependencyType && lhs.measurementID < rhs.measurementID);
+}
+bool operator>(const MeasurementDependency &lhs, const MeasurementDependency &rhs) {
+    return rhs < lhs;
+}
+bool operator<=(const MeasurementDependency &lhs, const MeasurementDependency &rhs) {
+    return !(rhs < lhs);
+}
+bool operator>=(const MeasurementDependency &lhs, const MeasurementDependency &rhs) {
+    return !(lhs < rhs);
+}
+bool operator==(const ReferencedImageSequence &lhs, const ReferencedImageSequence &rhs) {
+    return lhs.referencedSOPInstanceUID == rhs.referencedSOPInstanceUID;
+}
+bool operator!=(const ReferencedImageSequence &lhs, const ReferencedImageSequence &rhs) {
+    return !(rhs == lhs);
+}
+bool operator==(const MeasurementInformation &lhs, const MeasurementInformation &rhs) {
```

Comment:
This line and similar below are very long, hard to read. The ones above (e.g. `StudyInformation`) are much cleaner.